app-modules/ai/tests/Tenant/Feature/Jobs/CloneAiThreadTest.php

```diff
@@ -34,4 +34,48 @@
 </COPYRIGHT>
 */
 
-todo('can clone a thread and its messages');
+use AdvisingApp\Ai\Enums\AiApplication;
+use AdvisingApp\Ai\Enums\AiModel;
+use AdvisingApp\Ai\Jobs\CloneAiThread;
+use AdvisingApp\Ai\Models\AiAssistant;
+use AdvisingApp\Ai\Models\AiMessage;
+use AdvisingApp\Ai\Models\AiThread;
+use AdvisingApp\Authorization\Enums\LicenseType;
+use App\Models\User;
+use Filament\Notifications\DatabaseNotification;
+use Illuminate\Support\Facades\Notification;
+
+use function Pest\Laravel\withoutMiddleware;
+
+it('can clone a thread and its messages', function () {
+    Notification::fake();
+    withoutMiddleware();
+
+    $sender = User::factory()->licensed(LicenseType::cases())->create();
+    $recipient = User::factory()->licensed(LicenseType::cases())->create();
+
+    $assistant = AiAssistant::factory()->create([
+        'application' => AiApplication::Test,
+        'is_default' => true,
+        'model' => AiModel::Test,
+    ]);
+
+    $thread = AiThread::factory()
+        ->for($assistant, 'assistant')
+        ->has(AiMessage::factory()->count(3), 'messages')
+        ->for($sender, 'user')
+        ->create();
+
+    dispatch(new CloneAiThread($thread, $sender, $recipient));
```

Comment:
Rather than `dispatch` this it can be better to just new up the job class and trigger the handle method directly.

As the purpose of this test is to specifically test the functionality of this job.

Reply:
I actually suggested this approach, as calling the method directly would force you to handle dependency injection manually

Reply:
Hmm, alright. That's fine then. No need to change.